File: src/BuildingBlocks/Heyer.BuildingBlocks.Application/HttpLanguage/ListResponse.cs
namespace Heyer.BuildingBlocks.Application.HttpLanguage;

public record ListResponse<T>(int PageSize, long TotalCount, IEnumerable<T> Items)
{
    public static ListResponse<T> Create(IEnumerable<T> items, int pageSize, long totalCount) =>
        new(pageSize, totalCount, items);
}

File: web/backoffice/src/app/models/list-response.ts
export class ListResponse<T> {
  constructor(
    public PageSize: number,
    public TotalCount: number,
    public Items: T[],
  ) {}

  public static from<T>(response: ListResponse<T>): ListResponse<T> {
    const x = new ListResponse<T>(
      response.PageSize,
      response.TotalCount,
      response.Items,
    );

    console.log(x);

    return x;
  }
}
